File: dataset/codenet/Python/Code/s951169509.py
from collections import deque
H, W = map(int, input().split())
G = []
for i in range(H+2):
    if i == 0 or i == H+1:
        G.append('#'*(W+2))
    else:
        G.append('#' + input() + '#')

ans = 0
for sx in range(W+2):
    for sy in range(H+2):
        if G[sy][sx] == '.':
            dist = [[-float('inf') for i in range(W+2)] for i in range(H+2)]
            visited = [[-1 for i in range(W+2)] for i in range(H+2)]
            q = deque([])
            q.append((sx, sy))
            dist[sy][sx] = 0
            visited[sy][sx] = 1
            while q:
                x, y = q.popleft()

                if G[y][x+1] == '.' and visited[y][x+1] == -1:
                    q.append((x+1, y))
                    dist[y][x+1] = dist[y][x] + 1
                    visited[y][x+1] = 1

                if G[y][x-1] == '.' and visited[y][x-1] == -1:
                    q.append((x-1, y))
                    dist[y][x-1] = dist[y][x] + 1
                    visited[y][x-1] = 1

                if G[y+1][x] == '.' and visited[y+1][x] == -1:
                    q.append((x, y+1))
                    dist[y+1][x] = dist[y][x] + 1
                    visited[y+1][x] = 1

                if G[y-1][x] == '.' and visited[y-1][x] == -1:
                    q.append((x, y-1))
                    dist[y-1][x] = dist[y][x] + 1
                    visited[y-1][x] = 1
            for i in range(W+2):
                for j in range(H+2):
                    ans = max(ans, dist[j][i])
print(ans)

File: output/deepseek_chat/codenet/algo_based_translation/python/java/successful_translation/s951169509.java

import java.util.*;
import java.io.*;

public class s951169509 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] dimensions = br.readLine().split(" ");
        int H = Integer.parseInt(dimensions[0]);
        int W = Integer.parseInt(dimensions[1]);
        
        List<String> G = new ArrayList<>();
        for (int i = 0; i <= H + 1; i++) {
            if (i == 0 || i == H + 1) {
                G.add("#".repeat(W + 2));
            } else {
                G.add("#" + br.readLine() + "#");
            }
        }
        
        int ans = 0;
        for (int sx = 0; sx <= W + 1; sx++) {
            for (int sy = 0; sy <= H + 1; sy++) {
                if (G.get(sy).charAt(sx) == '.') {
                    double[][] dist = new double[H + 2][W + 2];
                    int[][] visited = new int[H + 2][W + 2];
                    for (int i = 0; i <= H + 1; i++) {
                        Arrays.fill(dist[i], Double.NEGATIVE_INFINITY);
                        Arrays.fill(visited[i], -1);
                    }
                    
                    Deque<int[]> q = new ArrayDeque<>();
                    q.add(new int[]{sx, sy});
                    dist[sy][sx] = 0;
                    visited[sy][sx] = 1;
                    
                    while (!q.isEmpty()) {
                        int[] current = q.poll();
                        int x = current[0];
                        int y = current[1];
                        
                        if (G.get(y).charAt(x + 1) == '.' && visited[y][x + 1] == -1) {
                            q.add(new int[]{x + 1, y});
                            dist[y][x + 1] = dist[y][x] + 1;
                            visited[y][x + 1] = 1;
                        }
                        
                        if (G.get(y).charAt(x - 1) == '.' && visited[y][x - 1] == -1) {
                            q.add(new int[]{x - 1, y});
                            dist[y][x - 1] = dist[y][x] + 1;
                            visited[y][x - 1] = 1;
                        }
                        
                        if (G.get(y + 1).charAt(x) == '.' && visited[y + 1][x] == -1) {
                            q.add(new int[]{x, y + 1});
                            dist[y + 1][x] = dist[y][x] + 1;
                            visited[y + 1][x] = 1;
                        }
                        
                        if (G.get(y - 1).charAt(x) == '.' && visited[y - 1][x] == -1) {
                            q.add(new int[]{x, y - 1});
                            dist[y - 1][x] = dist[y][x] + 1;
                            visited[y - 1][x] = 1;
                        }
                    }
                    
                    for (int i = 0; i <= W + 1; i++) {
                        for (int j = 0; j <= H + 1; j++) {
                            ans = Math.max(ans, (int) dist[j][i]);
                        }
                    }
                }
            }
        }
        
        System.out.println(ans);
    }
}

